TestFoundation/TestNSString.swift

```diff
@@ -1086,7 +1093,8 @@ let comparisonTests = [
     // ASCII cases
     ComparisonTest("t", "tt"),
     ComparisonTest("t", "Tt"),
-    ComparisonTest("\u{0}", ""),
+    ComparisonTest("\u{0}", "",
```

Comment:
Due to above change to checkHasPrefixHasSuffix ComparisonTest("\u{0}", "") fails on linux  because of https://bugs.swift.org/browse/SR-332 so modified this test with this reason.